### PR TITLE
Feature/create model

### PR DESCRIPTION
**Pull request by raodz**
## Description

- **Refactored Model Architecture:**
  - Consolidated the layers responsible for classification (excluding the `backbone`) into a single `self.classifier` attribute within the `BaldOrNotModel` class. This `classifier` now contains the `GlobalAveragePooling2D`, `Dense`, `Dropout` (if applicable), and final `Dense` layers with sigmoid activation. This change improves the modularity and clarity of the model's architecture by grouping all non-backbone layers into a single, cohesive block.

- **.gitignore Update:**
  - Added `*.pyc` to the `.gitignore` file to prevent issues related to committing Python bytecode files. This change was necessary to avoid conflicts and ensure a cleaner repository by ignoring these automatically generated files.

## Testing

- **Updated Tests:**
  - Modified existing tests to reflect the architectural change where layers are now encapsulated within `self.classifier`. These tests were updated to ensure that they properly validate the model's structure and behavior following the refactor.

- **Added New Tests:**
  - **Backbone Output Shape Test:** Introduced a new test to verify that the output of the `backbone` has the expected 4D shape, ensuring it is correctly formatted as `(batch_size, height, width, channels)`.
  - **Backbone to Classifier Compatibility Test:** Added a test to check that the output shape of the `backbone` correctly matches the input shape expected by the `classifier`. This ensures that the data passed from the `backbone` to the `classifier` is in the correct format, preventing potential runtime errors.

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Tests update
- [x] Documentation update
- [x] Changes that do not directly affect the source code or tests

## Checklist
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/model.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import tensorflow as tf
+from src.config_constants import DENSE_UNITS
+from src.constants import IMG_LEN, NUM_CHANNELS
+
+
+class BaldOrNotModel(tf.keras.Model):
+    """
+    Binary classification model to determine if a person in an image is bald
```
Comment (skrzypczykt) [inline src/model.py]: should be here or under init?

File: src/model.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            DENSE_UNITS, activation="relu"
+        )
+        self.dropout: tf.keras.layers.Layer | None = (
+            tf.keras.layers.Dropout(dropout_rate)
+            if dropout_rate is not None
+            else None
+        )
+        self.predictions: tf.keras.layers.Layer = tf.keras.layers.Dense(
+            1, activation="sigmoid"
+        )
```
Comment (skrzypczykt) [inline src/model.py]: Make it a `self.classifier = Sequential([...])`

**Review by skrzypczykt — COMMENTED** [review]
:)

File: src/model.py (hunk 1 above)
Comment (raodz) [reply src/model.py]: As I see here https://www.programiz.com/python-programming/docstrings docstring should be under class. There could be another one under init but in our case it would be nearly identical

Comment (skrzypczykt) [reply src/model.py]: OK, cool

File: src/model.py [hunk 3]
```diff
@@ -12,46 +12,34 @@ class BaldOrNotModel(tf.keras.Model):
...
-            DENSE_UNITS, activation="relu"
-        )
-        self.dropout: tf.keras.layers.Layer | None = (
-            tf.keras.layers.Dropout(dropout_rate)
-            if dropout_rate is not None
-            else None
-        )
-        self.predictions: tf.keras.layers.Layer = tf.keras.layers.Dense(
-            1, activation="sigmoid"
+        self.classifier: tf.keras.Sequential = tf.keras.Sequential()
```
Comment (skrzypczykt) [inline src/model.py:34]: Adding type hint here looks funny

**Review by skrzypczykt — APPROVED** [review]
I think it looks really nice!!

**Review by jakub1090cn — APPROVED** [review]
looks good
